Add new `System.Diagnostics.UnreachableException` polyfill

0xced opened the pull request:
### Closes #60

### Additional context

The `System.Diagnostics.UnreachableException.cs` file was not actually auto-generated (see #88).

Also, only the English localization was considered for the exception message.

0xced committed:
Add new `System.Diagnostics.UnreachableException` polyfill

The System.Diagnostics.UnreachableException.cs file was not actually auto-generated (see #88).

Fixes #60

0xced committed:
Add new `System.Diagnostics.UnreachableException` polyfill

The System.Diagnostics.UnreachableException.cs file was not actually auto-generated (see #88).

Fixes #60

cremor commented on src/PolySharp.SourceGenerators/EmbeddedResources/System.Diagnostics.UnreachableException.cs:
```diff
@@ -0,0 +1,42 @@
+    /// <summary>
+    /// Exception thrown when the program executes an instruction that was thought to be unreachable.
+    /// </summary>
+    public sealed class UnreachableException : Exception
```
I think this needs an `ExcludeFromCodeCoverage` attribute. See #63

0xced replied:
Rebased and addressed in a2c4dd2edb0d8a6053080d78902558a7c20e3243

cremor commented on src/PolySharp.SourceGenerators/EmbeddedResources/System.Diagnostics.UnreachableException.cs:
```diff
@@ -0,0 +1,42 @@
+    /// <summary>
+    /// Exception thrown when the program executes an instruction that was thought to be unreachable.
+    /// </summary>
+    public sealed class UnreachableException : Exception
```
Should this be `internal` by default? See #92 for a similar case.

0xced replied:
Rebased and addressed in a2c4dd2edb0d8a6053080d78902558a7c20e3243

0xced commented:
I just rebased on develop with a few improvements.

* Addressed feedback from @cremor
  * Added the `ExcludeFromCodeCoverage` attribute
  * Made the `UnreachableException` class `internal` instead of `public`
* Added mention of this new class in the README

0xced commented:
Is there anything holding this pull request @Sergio0694? It looks like several people are interested in it.

0xced commented:
I see there's some recent activity here in #127. Please don't forget about this pull request too if you're goint to release a new version of PolySharp. 😉

0xced commented:
`UnreachableException` is available in .NET 8 and .NET 9, the only .NET versions which are [still supported](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core).

But it still makes sense when targeting [.NET Standard](https://learn.microsoft.com/en-us/dotnet/standard/net-standard).

Any chance to get this merged, @Sergio0694?

Youssef1313 commented:
Ping @Sergio0694
